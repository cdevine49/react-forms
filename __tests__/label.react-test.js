import React from 'react';
import Label from '../lib/components/label';
import renderer from 'react-test-renderer';

describe("Label", () => {
  test("does not render without child", () => {
    const tree = renderer.create(
      <Label
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders with child", () => {
    const tree = renderer.create(
      <Label>Test Label</Label>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("adds asterisk if required", () => {
    const tree = renderer.create(
      <Label required>Required Field</Label>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("accepts props", () => {
    const tree = renderer.create(
      <Label
        htmlFor="htmlFor"
        id="labelId"
        className="label-class test"
        >
        Test Input
      </Label>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});