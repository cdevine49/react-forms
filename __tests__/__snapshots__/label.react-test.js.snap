// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label accepts props 1`] = `
<label
  className="label-class test"
  htmlFor="htmlFor"
  id="labelId"
>
  Test Input
</label>
`;

exports[`Label adds asterisk if required 1`] = `
<label>
  Required Field
  <span
    className="required"
  >
    *
  </span>
</label>
`;

exports[`Label does not render without child 1`] = `null`;

exports[`Label renders with child 1`] = `
<label>
  Test Label
</label>
`;
